core/src/main/java/org/mule/runtime/core/internal/config/DefaultFeatureManagementService.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.config;
+
+import static org.mule.runtime.feature.internal.togglz.config.MuleTogglzFeatureFlaggingUtils.getFeature;
+import static org.mule.runtime.feature.internal.togglz.config.MuleTogglzFeatureFlaggingUtils.setFeatureState;
+import static org.mule.runtime.feature.internal.togglz.config.MuleTogglzFeatureFlaggingUtils.withFeatureUser;
+
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.feature.api.management.FeatureFlaggingManagementService;
+import org.mule.runtime.feature.internal.togglz.user.MuleTogglzArtifactFeatureUser;
+import org.togglz.core.repository.FeatureState;
+
+import javax.inject.Inject;
+
+/**
+ * The default implementation for a service to enable/disable runtime features.
+ *
+ * @since 4.5.0
+ */
+public class DefaultFeatureManagementService implements FeatureFlaggingManagementService {
```

Comment:
can this be moved to `mule-module-feature-management `?

Reply:
Created ticket MULE-19954. 
The idea is to make feature flagging fully decoupled from the core module.
It should be injectable as a service through an interface. Possibly the managed service should be at the container level.
This is currently built and added to the TransientRegistry manually when the muleContext is created.